File: sortingAlgorithms/selectionSort/selectionsort.cpp
#include <iostream>
#include <iterator>
#include <array>

using namespace std;

int arr[] = {2,1,7,4,3,8,66,11};

void selectionsort(int arr[]);


int main(){

	selectionsort(arr);

	for(int i = 0; i < 8; i++){

		cout << arr[i] << " " ;

	}

	cout << endl;

}

void selectionsort(int arr[]){

	int min;
	for(int i = 0 ; i < 8 ; i++){

		min = i;

		for(int j = min ; j < 8 ; j++){

			if(arr[j] < arr[min]){

				arr[j] = arr[min] + arr[j];
				arr[min] = arr[j] - arr[min];
				arr[j] = arr[j] - arr[min];	

			}

		}

	}

}

File: sortingAlgorithms/selectionSort/selection_sort.go
package main

import "fmt"

var arr = []int{5,8,6,9,4,7}

func main(){

	selectionsort(arr)
	fmt.Println(arr)
}


func selectionsort(arr []int) {
	
 	var minvalueidex int = 0

 	for j := 0; j < len(arr)-1; j++ {

 		for i := minvalueidex+1; i < len(arr)-1; i++ {
 		if arr[i]<arr[minvalueidex] {
 			var temp int = arr[i]
 				arr[i]=arr[minvalueidex]
 				arr[minvalueidex]=temp
 				minvalueidex++
 			}
		
 		}
 	}
 }


